SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -939,6 +939,25 @@ - (void)sendMessage:(TSOutgoingMessage *)message
         }
     }
 
+    NSString *localNumber = [TSAccountManager localNumber];
+    if ([localNumber isEqualToString:recipient.uniqueId]) {
+        if (deviceMessages.count < 1) {
+            DDLogInfo(@"Ignoring sync message without linked devices: %@", [message class]);
+            OWSAssert([message isKindOfClass:[OWSOutgoingSyncMessage class]]);
+
+            dispatch_async([OWSDispatch sendingQueue], ^{
+                [recipient save];
+                [self handleMessageSentLocally:message];
```

Comment:
If this code is about preventing sync-message sending - presumably it was called via `sendSyncTranscriptForMessage` which in turn is only ever called from within `handleMessageSentLocally`.

Currently  this shouldn't cause an infinite loop because `handleMessageSentLocally` has a short-circuit to make sure it only syncs once, but why call it at all here?

Reply:
I think you're right - this should be harmless but always redundant.  Removed.